cli/azd/docs/experimental-features.md

```diff
@@ -0,0 +1,54 @@
+# Experimental Features
+
+The Azure Developer CLI includes a `feature toggle` which allows the cli to ship `in-preview` ( _a.k.a. experimental_ ). This document provides the feature design for it.
+
+## Main requirement
+
+As a customer, I can explicitly ask `azd` to unblock and show any _experimental_ functionality. The features becoming available are not finalized, so they can be changed or removed in future releases.
+
+The strategy to ask `azd` to `toggle` between modes is:
+
+- Enable _all_ experimental features: `azd config set experimental on`
+- Enable _specific_ experimental feature: `azd config set experimental.name.feature`
+- Disable _all_ experimental features: `azd config unset experimental`
+- Disable _specific_ experimental feature: `azd config unset experimental.name.feature`
+
+## Experimental features
+
+The Azure Developer CLI offers the following type of features that can be toggled using experimental modes:
```

Comment:
It might also be worth noting that in the future/once we get this enabled, all features will start in alpha and move through the graduation system to stable.

Reply:
I'd actually probably consider leaving this section to a separate section that outline the phases, what they mean and what it takes to graduate.